src/main/kotlin/app/revanced/patches/youtube/ad/general/bytecode/patch/GeneralBytecodeAdsPatch.kt

```diff
@@ -248,22 +249,27 @@ class GeneralBytecodeAdsPatch : BytecodePatch() {
                                 }
 
                                 resourceIds[3], resourceIds[4], resourceIds[5] -> { // end screen ads
-                                    //  and is followed by an instruction with the mnemonic IPUT_OBJECT
-                                    val insertIndex = index + 7
-                                    val invokeInstruction = instructions.elementAt(insertIndex)
-                                    if (invokeInstruction.opcode != Opcode.IPUT_OBJECT) return@forEachIndexed
+                                    //  and is followed by an instruction with the mnemonic CHECK_CAST
+                                    var insertIndex = index + 4
```

Comment:
Use a subList view of the current instruction list to find the insertion index instead of checking two offsets like this.